New page navigation based on JS

tochman opened the pull request:
## PT Link

https://www.pivotaltracker.com/story/show/161696353

## PR Includes:

* modifies the create rsc link to ajax and adds method to display a pop-up
* login and sign up using JavaScript 

![rsc_js_flow](https://user-images.githubusercontent.com/5248170/47950721-91236e80-df56-11e8-97e1-5af8b3a37542.gif)

tochman committed:
modifies the create rsc link to ajax and adds method to display a pop-up

tochman committed:
adds function to hide pop-uo and show main content again

tochman committed:
refactors scenario to close pop-up before we click logout

tochman committed:
merges with upstream and fixes merge conflicts

tochman committed:
refactors navigation for dashboard

tochman committed:
adds new versions of templates to cover both html and js requests

Prozak8 commented on app/views/jobs/index.js.erb:
```diff
@@ -0,0 +1,3 @@
+
+var templateContent = '<%= escape_javascript(render partial: 'index') %>'
```
This can be refactored to j(render partial: 'index')

tochman replied:
I know. I prefer to use the more verbose syntax in student projects.

tochman committed:
add js flow to devise registrations and sessions controllers
adds remote true to shared devise links together with a modal toggler function
adds :extras content to madal that can be used IF we are making a js call
WIP error messages and responses

tochman committed:
adds js based flow for logging in and signing up
adds a custop path name to devise urls to avoid conflicts with User#create actions

tochman committed:
removes pry call

tochman committed:
moves flash notifications into grid
